pkg/proc/variables.go

```diff
@@ -1220,28 +1223,51 @@ func extractVarInfoFromEntry(tgt *Target, bi *BinaryInfo, image *Image, regs op.
 	return v, nil
 }
 
-// If v is a pointer a new variable is returned containing the value pointed by v.
+// If v is a pointer, its child value is loaded and returned as a new variable.
```

Comment:
maybeDereference gets called in a lot of places, I'm not comfortable with changing its semantics so much like this, especially without chaning its name and signature.

Reply:
By "changing its semantics" you mean the fact that `maybeDereference` now mutates `v`?
Or the fact that this part is gone?
```
if v.Unreadable != nil {
   return v
}
```

I've reverted this function to the how it was before, but I've had to change `evalPointerDeref()` to not use it any more, in order to avoid the bogus "nil pointer dereference" in the unreadable case. Please consider whether you're indeed happy with `maybeDereference()`, or whether some different semantics would actually be better for all callers.